server/server.go

```diff
@@ -766,6 +766,11 @@ func (s *Server) createClient(conn net.Conn) *client {
 		return c
 	}
 
+	// If there is a max subscriptions specified, add to the client.
```

Comment:
Should this be moved down (or up) to under the client lock instead of server lock?

Reply:
Will fix..